### DISCUSSION TITLE
Using with GAPI

### DISCUSSION THREAD

**Discussion by lucasow** [wavezync/vue3-google-signin#49] [Q&A] [answered]
Hi,

Iam using vue-gapi for calling App scripts and Drive API.
As GAPI auth2 is depracated now I would like to start using vue3-google-signin for autorization.
But I am still confused how to still use GAPI while using vue3-google-signin for autorization.

Could someone give me a hint please.

Many thanks,
Lucas

**Comment by kasvith** [comment]
You will only need vue3-google-signin for the task :)

**Reply by lucasow** [reply]
Sorry, I was not specific with my answer. I know I can use useTokenClient and obtain access_token but I would like to use GoogleSignInButton component as I also want to use vue3-google-signin for logging users to my system. So I got CredentialResponse. But I dont know how to obtain access_token from this CredentialResponse.

**Reply by kasvith** [reply]
```vue
<script setup lang="ts">
import {
  GoogleSignInButton,
  type CredentialResponse,
} from "vue3-google-signin";

// handle success event
const handleLoginSuccess = (response: CredentialResponse) => {
  const { credential } = response;
  console.log("Access Token", credential);
};

// handle an error event
const handleLoginError = () => {
  console.error("Login failed");
};
</script>

<template>
  <GoogleSignInButton
    @success="handleLoginSuccess"
    @error="handleLoginError"
  ></GoogleSignInButton>
</template>
```

Like this https://vue3-google-signin.vercel.app/guide/google-signin-button.html#using-google-sign-in-button

**Reply by lucasow** [reply]
Thank you. 
But Iam still unable to call google apis with credential obtained.
There is my code

const handleLoginSuccess = (response) => {
                const { credential } = response;
                console.log("Access Token", credential);
                $.ajax({
                    url: "https://content-script.googleapis.com/v1/scripts/${scriptId}:run?alt=json&key=${secretKey}",
                    beforeSend: function(request) {
                        request.setRequestHeader("Authorization", "Bearer "+credential)
                    },
                    data: {
                        function: "test"
                    },
                    dataType: "json",
                    type: "POST",
                    success: function (response) {
                        console.log('response', response)
                    },

                    error: function ({responseJSON: error}) {
                        console.log(error);
                    }
                });
            };


I got returned: Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential.

If I use same code with access_token obtained with method useTokenClient, it works.
What am I missing?

**Reply by kasvith** [reply, marked as answer]
@lucasow no thats how it works, default button from google doesnt allow setting up scopes like drive access. you have to create a custom button for that

**Comment by lucasow** [comment]
> @lucasow no thats how it works, default button from google doesnt allow setting up scopes like drive access. you have to create a custom button for that

Many thanks for your answer. That information helps me to stop struggling with default Google button. So I will use custom button. :)
